docs/src/snap/howto/epa.md

```diff
@@ -0,0 +1,1061 @@
+
+```
+git clone https://github.com/DPDK/dpdk.git /home/ubuntu/dpdk
+cat <<EOF > /var/lib/cloud/scripts/per-boot/dpdk_bind.sh 
+      #!/bin/bash
+      if [ -d /home/ubuntu/dpdk ]; then
+        modprobe vfio-pci
+        vfs=$(python3 /home/ubuntu/dpdk/usertools/dpdk-devbind.py -s | grep drv=iavf | awk '{print $1}' | tail -n +11)
+        python3 /home/ubuntu/dpdk/usertools/dpdk-devbind.py --bind=vfio-pci $vfs
+      fi
+sudo chmod 0755 /var/lib/cloud/scripts/per-boot/dpdk_bind.sh
+```
+```{dropdown} Explanation 
+  * Load VFIO Module (modprobe vfio-pci): If the DPDK directory exists, the script loads the VFIO-PCI kernel module. This module is necessary for the VFIO driver to function.  
+  * The script uses the dpdk-devbind.py tool (included with DPDK) to list the available network devices and their drivers.  
+    * It filters this output using grep drv=iavf to find devices that are currently using the iavf driver (a common driver for Intel network adapters), excluding the physical network interface itself and just focusing on the virtual functions (VFs).  
+  * Bind VFs to VFIO: The script uses dpdk-devbind.py again, this time with the \--bind=vfio-pci option, to bind the identified VFs to the VFIO-PCI driver. This step essentially tells the kernel to relinquish control of these devices to DPDK.  
+```
+
+To test that the VFIO Kernel Module and DPDK are enabled:
+
+```
+lsmod | grep -E 'vfio'
+```
+
+...should indicate the kernel module is loaded
+
+```
+vfio_pci               16384  0
+vfio_pci_core          94208  1 vfio_pci
+vfio_iommu_type1       53248  0
+vfio                   73728  3 vfio_pci_core,vfio_iommu_type1,vfio_pci
+iommufd                98304  1 vfio
+irqbypass              12288  2 vfio_pci_core,kvm
+
+```
+
+Running the helper script:
+
+```
+python3 /home/ubuntu/dpdk/usertools/dpdk-devbind.py -s
+```
+
+...should return a list of network devices using DPDK:
+
+```
+Network devices using DPDK-compatible driver
+============================================
+0000:98:12.2 'Ethernet Adaptive Virtual Function 1889' drv=vfio-pci unused=iavf
+0000:98:12.3 'Ethernet Adaptive Virtual Function 1889' drv=vfio-pci unused=iavf
+0000:98:12.4 'Ethernet Adaptive Virtual Function 1889' drv=vfio-pci unused=iavf
+....
+```
+
+With these preparation steps we have enabled the features of EPA:
+
+* NUMA and CPU Pinning are available to the first 32 CPUs  
+* Real-Time Kernel is enabled  
+* HugePages are enabled and 1000 1G huge pages are available   
+* SRIOV is enabled in the enp152s0f1 interface, with 128 virtual function interfaces bound to the vfio-pci driver (that could also use the iavf driver)  
+* DPDK is enabled in all the 128 virtual function interfaces
+
+````
+
+````{group-tab} MAAS
+
+To prepare a machine for CPU isolation, Hugepages, real-time kernel, SRIOV and DPDK we leverage cloud-init through MAAS.
+
+```
+#cloud-config
```

Comment:
I thought I had replied to this one? I think that is an excellent idea, but i think that applies to all our docs, and I think it would make sense to make a card specifically for finding and adding download links.